From: Jens Axboe <axboe@fb.com>
Date: Fri, 6 Mar 2015 12:56:13 -0700
Subject: NVMe: increase depth of admin queue
References: bnc#936076,FATE#318544
Patch-Mainline: v4.1-rc1
Git-commit: d31af0a325ca4ff8e57b8616ab2228913df369ad)

Usually the admin queue depth of 64 is plenty, but for some use cases we
really need it larger. Examples are use cases like MAT, where you have
to touch all of NAND for init/format like purposes. In those cases, we
see a good 2x increase with an increased queue depth.

[hare: Ported to SLE12 SP1]

Signed-off-by: Jens Axboe <axboe@fb.com>
Acked-by: Keith Busch <keith.busch@intel.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/block/nvme-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/block/nvme-core.c b/drivers/block/nvme-core.c
index 7e3b480..728c0d2 100644
--- a/drivers/block/nvme-core.c
+++ b/drivers/block/nvme-core.c
@@ -1572,7 +1572,7 @@ static int nvme_configure_admin_queue(struct nvme_dev *dev)
 
 	nvmeq = raw_nvmeq(dev, 0);
 	if (!nvmeq) {
-		nvmeq = nvme_alloc_queue(dev, 0, 64, 0);
+		nvmeq = nvme_alloc_queue(dev, 0, 256, 0);
 		if (!nvmeq)
 			return -ENOMEM;
 	}
-- 
1.8.5.2

